File: falling/submissions/accepted/falling_lukas.cs
//Solution by lukasP (Lukáš Poláček)
using System;
using System.Linq;

class Numbertree
{
    public static void Main()
    {
        int d = int.Parse(Console.ReadLine());
        if (d % 4 == 0)
        {
            int n = d / 4 - 1;
            Console.WriteLine(n + " " + (n + 2));
        }
        else if ((d & 1) == 1)
        {
            int n = d / 2;
            Console.WriteLine(n + " " + (n + 1));
        }
        else
        {
            Console.WriteLine("impossible");
        }
    }
}

File: falling/submissions/accepted/falling_lukas.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-
# Solution by lukasP (Lukáš Poláček)
from sys import stdin, exit

d = int(stdin.readline())

for i in range(0, 200000):
    si = i * i
    for j in range(i + 1, 200000):
        sj = j * j
        if sj - si > d:
            break
        if sj - si == d:
            print("{0} {1}".format(i, j))
            exit(0)

print("impossible")
